alfresco-java-rest-api-common/src/main/java/org/alfresco/rest/sdk/feign/oauth2/OAuth2FeignRequestInterceptor.java

```diff
@@ -55,7 +57,36 @@ public void apply(RequestTemplate template) {
     }
 
     private String getAuthorizationToken() {
-        final OAuth2AccessToken accessToken = oAuth2AuthorizedClientManager.authorize(oAuth2AuthorizeRequest).getAccessToken();
+        OAuth2AccessToken accessToken = null;
+        try {
+            accessToken = authorizeAndGetToken();
+        } catch (OAuth2AuthorizationException e) {
+            if (isTokenExpired(e)) {
+                LOGGER.info("Reauthorization required: " + e.getMessage());
+                accessToken = reauthorize();
+            } else {
+                LOGGER.error("Authorization failed: " + e.getMessage());
+            }
+        }
         return String.format(AUTH_HEADER_FORMAT, accessToken.getTokenType().getValue(), accessToken.getTokenValue());
     }
+
+    // Reauthorize and get a new token
+    private OAuth2AccessToken reauthorize() {
+        return authorizeAndGetToken();
+    }
+
+    // Authorize and get token
+    private OAuth2AccessToken authorizeAndGetToken() {
+        OAuth2AuthorizedClient authorizedClient = oAuth2AuthorizedClientManager.authorize(oAuth2AuthorizeRequest);
+        if (authorizedClient == null || authorizedClient.getAccessToken() == null) {
+            LOGGER.error("Failed to authorize and get access token.");
+        }
+        return authorizedClient.getAccessToken();
+    }
+
+    private boolean isTokenExpired(OAuth2AuthorizationException e) {
+        return e.getMessage().contains("Token is not active");
```

Comment:
The way to identify the token is expired i.e. decoding the error message and then identifying the token is expired is wrong. To determine the expiration time of the current JWT token that was created, you can decode the token and check the value of the “exp” claim. If the expiration time is exceeding the current time then it means that the token is expired.

Reply:
Made changes to validate the access token and refresh token by comparing the expiration time.